### PR TITLE
Add is_patched to security_scan

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/ralph/security/migrations/0007_auto_20170331_0934.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('security', '0006_securityscan_is_patched'),
+    ]
+
+    operations = [
+        migrations.RunPython(update_is_patched, reverse_func),
```
Comment (romcheg) [inline src/ralph/security/migrations/0007_auto_20170331_0934.py]: According to the documentation the **reverse_code** parameter is optional. You can remove the entire reverse_func from this file, if there's not going to be any implementation.

File: src/ralph/security/migrations/0007_auto_20170331_0934.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
```
Comment (romcheg) [inline src/ralph/security/migrations/0007_auto_20170331_0934.py:1]: Does it make sense to merge this migration with 0006_securityscan_is_patched?

File: src/ralph/security/migrations/0007_auto_20170331_0934.py (hunk 1 above)
Comment (romcheg) [reply src/ralph/security/migrations/0007_auto_20170331_0934.py]: I would also add a migration that sets the nullable property to False after this one.

File: src/ralph/security/migrations/0007_auto_20170331_0934.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        vulnerability__patch_deadline__gt=datetime.now()
+    ).values_list(
+        'securityscan_id', flat=True
+    ).distinct()
+
+    patched_scans = SecurityScan.objects.filter(id__in=is_patched_ids).update(
+        is_patched=True
+    )
+
+    print()
```
Comment (romcheg) [inline src/ralph/security/migrations/0007_auto_20170331_0934.py]: Was this empty print left intentionally? Not an issue, just checking :)

File: src/ralph/security/models.py [hunk 4]
```diff
@@ -82,6 +86,11 @@ class SecurityScan(
         on_delete=models.CASCADE,
     )
     vulnerabilities = models.ManyToManyField(Vulnerability, blank=True)
+    # this is a quirk field, it is updated manually (for now it's in API)
+    # this is because it's hard to handling it automatically
+    # (its value is computated depending on M2M field and M2M signals are
+    # complicated)
+    is_patched = models.BooleanField(default=False)
```
Comment (romcheg) [inline src/ralph/security/models.py:93]: Perhaps it makes sense to make it non-nullable, doesn't it?

**Review by romcheg — CHANGES_REQUESTED** [review]
A few small things. Otherwise LGTM.

File: src/ralph/security/migrations/0007_auto_20170331_0934.py (hunk 3 above)
Comment (ghost) [reply src/ralph/security/migrations/0007_auto_20170331_0934.py]: yup

File: src/ralph/security/migrations/0007_auto_20170331_0934.py (hunk 2 above)
Comment (ghost) [reply src/ralph/security/migrations/0007_auto_20170331_0934.py:1]: 0007 is custom migrations. i'd like to have them seperated unless there is a reason.
Is there a reason? :)

File: src/ralph/security/migrations/0007_auto_20170331_0934.py (hunk 1 above)
Comment (ghost) [reply src/ralph/security/migrations/0007_auto_20170331_0934.py]: If I remove it,  revers migration will be still possible?

File: src/ralph/security/models.py (hunk 4 above)
Comment (ghost) [reply src/ralph/security/models.py:93]: I think it is.
https://docs.djangoproject.com/en/1.8/ref/models/fields/#django.db.models.BooleanField

> If you need to accept null values then use NullBooleanField instead.

File: src/ralph/security/migrations/0007_auto_20170331_0934.py (hunk 1 above)
Comment (romcheg) [reply src/ralph/security/migrations/0007_auto_20170331_0934.py]: No, but there is a special noop code for this available as [RunPython.noop](https://docs.djangoproject.com/en/1.11/ref/migration-operations/#django.db.migrations.operations.RunPython.noop)

File: src/ralph/security/migrations/0007_auto_20170331_0934.py (hunk 2 above)
Comment (romcheg) [reply src/ralph/security/migrations/0007_auto_20170331_0934.py:1]: No reason IIRC except personal preferences actually. Let's leave it as is it then.

File: src/ralph/security/migrations/0007_auto_20170331_0934.py (hunk 1 above)
Comment (ghost) [reply src/ralph/security/migrations/0007_auto_20170331_0934.py]: I don't see example of using it so, I prefere my version which it :sparkles: just-works :sparkles: 
anyway, when I'm using it like this:

```
      operations = [                                                            
          migrations.RunPython(update_is_patched, migrations.RunSQL.noop),         
      ] 
```

i get:

`ValueError: RunPython must be supplied with callable arguments `

File: src/ralph/security/models.py (hunk 4 above)
Comment (romcheg) [reply src/ralph/security/models.py:93]: Ok, makes sense

File: src/ralph/security/migrations/0007_auto_20170331_0934.py (hunk 1 above)
Comment (romcheg) [reply src/ralph/security/migrations/0007_auto_20170331_0934.py]: That's weird... let's leave it as is then.

**Review by romcheg — APPROVED** [review]

File: src/ralph/security/migrations/0007_auto_20170331_0934.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+def update_is_patched(apps, schema_editor):
+    SecurityScan = apps.get_model("security", "SecurityScan")
+    is_patched_ids = SecurityScan.vulnerabilities.through.objects.filter(
+        vulnerability__patch_deadline__gt=datetime.now()
+    ).values_list(
+        'securityscan_id', flat=True
+    ).distinct()
+
+    patched_scans = SecurityScan.objects.filter(id__in=is_patched_ids).update(
```
Comment (mkurek) [inline src/ralph/security/migrations/0007_auto_20170331_0934.py]: in this case, when my sec scan has two vulnerabilities - one with deadline before today, and one with deadline after today, it'll be marked as is_patched, isn't it?

File: src/ralph/security/migrations/0007_auto_20170331_0934.py (hunk 1 above)
Comment (mkurek) [reply src/ralph/security/migrations/0007_auto_20170331_0934.py]: @xliiv `migrations.RunPython.noop` (not `migrations.RunSQL.noop`)

**Review by mkurek — CHANGES_REQUESTED** [review]

File: src/ralph/security/migrations/0007_auto_20170331_0934.py (hunk 5 above)
Comment (ghost) [reply src/ralph/security/migrations/0007_auto_20170331_0934.py]: now fixed

**Review by mkurek — APPROVED** [review]

**Pull request merged**